File: 217_ContainsDuplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        self.numsSet = set(nums)
        if len(nums) == len(self.numsSet):    ## 利用set容器的元素唯一性进行去重
            return False
        return True

File: 217_ContainsDuplicate.cpp
// Solution1
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        std::unordered_map<int, int> elements;
        int cnt = nums.size();
        for(int i = 0; i != cnt; ++i)
            ++elements[nums[i]];
        for(const auto &e : elements)
            if(e.second >= 2)
                return true;
        return false;
    }
};

// Solution2
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        set<int> iset(nums.cbegin(), nums.cend());
        return nums.size() != iset.size() ? true : false;
    }
};
